sat-plugin/src/main/java/org/openhab/tools/analysis/tools/PmdChecker.java

```diff
@@ -105,7 +152,34 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                 configuration, pmdPlugins);
 
         log.debug("PMD execution has been finished.");
-
     }
 
+    private void generateRulesXmlFile(String location) throws MojoExecutionException {
+        try {
+            String rulesLocation = getLocation(pmdRuleset, DEFAULT_RULESET_XML);
+            URL url = new URL(rulesLocation);
+            SAXReader reader = new SAXReader();
+            Document document = reader.read(url);
+            List<Node> ruleNodes = document.getRootElement().elements("rule");
+            ruleNodes.forEach(node -> {
+                // There is another Element type coming from
+                // org.twdata.maven.mojoexecutor.MojoExecutor.*
+                org.dom4j.Element nodeAsElement = (org.dom4j.Element) node;
+                String ruleType = nodeAsElement.attributeValue("ref");
+                for (String category : this.skippedCategories) {
+                    if (ruleType.contains(category)) {
+                        getLog().debug(String.format(CATEGORY_SKIP_MESSAGE, ruleType, category));
+                        // We detach the rule if it belongs to a skipped category
+                        node.detach();
+                    }
+                }
+            });
+
+            FileOutputStream stream = new FileOutputStream(location);
+            new XMLWriter(stream, OutputFormat.createPrettyPrint()).write(document);
+            stream.close();
```

Comment:
Are all streams properly closed in case of exceptions? Try-with-resources?

Should we close the reader or the document?

Of course also apply to the other one.